// @apps-in-toss/web-bridge/build/index.js
export * from './bridge.js'

// src/index.ts
export * from "@apps-in-toss/bridge-core";

// src/storage.ts
import { createAsyncBridge } from "@apps-in-toss/bridge-core";
var Storage = {
  /**
   * @public
   * @category 저장소
   * @name getItem
   * @description 모바일 앱의 로컬 저장소에서 문자열 데이터를 가져와요. 주로 앱이 종료되었다가 다시 시작해도 데이터가 유지되어야 하는 경우에 사용해요.
   * @param {string} key - 가져올 아이템의 키를 입력해요.
   * @returns {Promise<string | null>} 지정한 키에 저장된 문자열 값을 반환해요. 값이 없으면 `null`을 반환해요.
   * @example
   *
   * ### `my-key`에 저장된 아이템 가져오기
   * ```ts
   * const value = await Storage.getItem('my-key');
   * console.log(value); // 'value'
   * ```
   */
  getItem: createAsyncBridge("getStorageItem"),
  /**
   * @public
   * @category 저장소
   * @name setItem
   * @description 모바일 앱의 로컬 저장소에 문자열 데이터를 저장해요. 주로 앱이 종료되었다가 다시 시작해도 데이터가 유지되어야 하는 경우에 사용해요.
   * @param {string} key - 저장할 아이템의 키를 입력해요.
   * @param {string} value - 저장할 아이템의 값을 입력해요.
   * @returns {Promise<void>} 아이템을 성공적으로 저장하면 아무 값도 반환하지 않아요.
   * @example
   *
   * ### `my-key`에 아이템 저장하기
   * ```ts
   * import { Storage } from '@apps-in-toss/framework';
   *
   * await Storage.setItem('my-key', 'value');
   * ```
   */
  setItem: createAsyncBridge("setStorageItem"),
  /**
   * @public
   * @category 저장소
   * @name removeItem
   * @description 모바일 앱의 로컬 저장소에서 특정 키에 해당하는 아이템을 삭제해요.
   * @param {string} key - 삭제할 아이템의 키를 입력해요.
   * @returns {Promise<void>} 아이템을 삭제하면 아무 값도 반환하지 않아요.
   * @example
   *
   * ### `my-key`에 저장된 아이템 삭제하기
   * ```ts
   * import { Storage } from '@apps-in-toss/framework';
   *
   * await Storage.removeItem('my-key');
   * ```
   */
  removeItem: createAsyncBridge("removeStorageItem"),
  /**
   * @public
   * @category 저장소
   * @name clearItems
   * @description 모바일 앱의 로컬 저장소의 모든 아이템을 삭제해요.
   * @returns {Promise<void>} 아이템을 삭제하면 아무 값도 반환하지 않고 저장소가 초기화돼요.
   * @example
   *
   * ### 저장소 초기화하기
   * ```ts
   * import { Storage } from '@apps-in-toss/framework';
   *
   * await Storage.clearItems();
   * ```
   */
  clearItems: createAsyncBridge("clearItems")
};

// src/iap.ts
import { createAsyncBridge as createAsyncBridge2 } from "@apps-in-toss/bridge-core";
var IAP = {
  /**
   * @public
   * @category 인앱결제
   * @name iapCreateOneTimePurchaseOrder
   * @description
   * 특정 인앱결제 주문서 페이지로 이동해요. 사용자가 상품 구매 버튼을 누르는 상황 등에 사용할 수 있어요. 사용자의 결제는 이동한 페이지에서 진행돼요. 만약 결제 중에 에러가 발생하면 에러 유형에 따라 에러 페이지로 이동해요.
   * @param {IapCreateOneTimePurchaseOrderOptions} params - 인앱결제를 생성할 때 필요한 정보예요.
   * @param {string} params.productId - 주문할 상품의 ID예요.
   * @returns {Promise<IapCreateOneTimePurchaseOrderResult | undefined>} 결제에 성공하면 결제 결과 객체를 반환해요. 앱 버전이 최소 지원 버전(안드로이드 5.219.0, iOS 5.219.0)보다 낮으면 인앱결제를 실행할 수 없어서 `undefined`를 반환해요.
   *
   * @throw {code: "INVALID_PRODUCT_ID"} - 유효하지 않은 상품 ID이거나, 해당 상품이 존재하지 않을 때 발생해요.
   * @throw {code: "PAYMENT_PENDING"} - 사용자가 요청한 결제가 아직 승인을 기다리고 있을 때 발생해요.
   * @throw {code: "NETWORK_ERROR"} - 서버 내부 문제로 요청을 처리할 수 없을 때 발생해요.
   * @throw {code: "INVALID_USER_ENVIRONMENT"} - 특정 기기, 계정 또는 설정 환경에서 구매할 수 없는 상품일 때 발생해요.
   * @throw {code: "ITEM_ALREADY_OWNED"} - 사용자가 이미 구매한 상품을 다시 구매하려고 할 때 발생해요.
   * @throw {code: "APP_MARKET_VERIFICATION_FAILED"} - 사용자가 결제를 완료했지만, 앱스토어에서 사용자 정보 검증에 실패했을 때 발생해요. 사용자가 앱스토어에 문의해서 환불을 요청해야해요.
   * @throw {code: "TOSS_SERVER_VERIFICATION_FAILED"} - 사용자가 결제를 완료했지만, 서버 전송에 실패해서 결제 정보를 저장할 수 없을 때 발생해요.
   * @throw {code: "INTERNAL_ERROR"} - 서버 내부 문제로 요청을 처리할 수 없을 때 발생해요.
   * @throw {code: "KOREAN_ACCOUNT_ONLY"} - iOS 환경에서 사용자의 계정이 한국 계정이 아닐 때 발생해요.
   * @throw {code: "USER_CANCELED"} - 사용자가 결제를 완료하지 않고 주문서 페이지를 이탈했을 때 발생해요.
   */
  createOneTimePurchaseOrder: createAsyncBridge2(
    "iapCreateOneTimePurchaseOrder"
  ),
  /**
   * @public
   * @category 인앱결제
   * @name iapGetProductItemList
   * @description 인앱결제로 구매할 수 있는 상품 목록을 가져와요. 상품 목록 화면에 진입할 때 호출해요.
   * @returns {Promise<{ products: IapProductListItem[] } | undefined>} 상품 목록을 포함한 객체를 반환해요. 앱 버전이 최소 지원 버전(안드로이드 5.219.0, iOS 5.219.0)보다 낮으면 `undefined`를 반환해요.
   */
  getProductItemList: createAsyncBridge2("iapGetProductItemList")
};

// src/getSafeAreaInsets.ts
import { createConstantBridge } from "@apps-in-toss/bridge-core";
var getSafeAreaBottom = createConstantBridge("getSafeAreaBottom");
var getSafeAreaTop = createConstantBridge("getSafeAreaTop");
function getSafeAreaInsets() {
  return { top: getSafeAreaTop(), bottom: getSafeAreaBottom() };
}

// src/googleAdMob.ts
import { createConstantBridge as createConstantBridge2, createEventBridge } from "@apps-in-toss/bridge-core";
var GoogleAdMob = {
  /**
   * @public
   * @category 광고
   * @name loadAdMobInterstitialAd
   * @description 앱 화면 전체를 덮는 전면 광고를 미리 불러와서, 광고가 필요한 시점에 바로 보여줄 수 있도록 준비하는 함수예요.
   * @param {LoadAdMobInterstitialAdParams} params 광고를 불러올 때 사용할 설정 값이에요. 광고 ID와 광고의 동작에 대한 콜백을 설정할 수 있어요.
   * @param {LoadAdMobInterstitialAdOptions} params.options 광고를 불러올 때 전달할 옵션 객체예요.
   * @param {string} params.options.adUnitId 광고 단위 ID예요. 발급받은 전면 광고용 ID를 입력해요.
   * @param {(event: LoadAdMobInterstitialAdEvent) => void} [params.onEvent] 광고 관련 이벤트가 발생했을 때 호출돼요. (예시: 광고가 닫히거나 클릭됐을 때). 자세한 이벤트 타입은 [LoadAdMobInterstitialAdEvent](/react-native/reference/framework/광고/LoadAdMobInterstitialAdEvent.html)를 참고하세요.
   * @param {(reason: unknown) => void} [params.onError] 광고를 불러오지 못했을 때 호출돼요. (예시: 네트워크 오류나 지원하지 않는 환경일 때)
   * @property {() => boolean} [isSupported] 현재 실행 중인 앱(예: 토스 앱, 개발용 샌드박스 앱 등)에서 Google AdMob 광고 기능을 지원하는지 확인하는 함수예요. 기능을 사용하기 전에 지원 여부를 확인해야 해요.
   *
   * @example
   * ### 뷰 진입 시 전면광고 불러오기
   * ```tsx
   * import { GoogleAdMob } from '@apps-in-toss/framework';
   * import { useEffect } from 'react';
   * import { View, Text } from 'react-native';
   *
   * const AD_UNIT_ID = '<YOUR_AD_UNIT_ID>';
   *
   * function Page() {
   *   useEffect(() => {
   *     if (GoogleAdMob.loadAdMobInterstitialAd.isSupported() !== true) {
   *       return;
   *     }
   *
   *     const cleanup = GoogleAdMob.loadAdMobInterstitialAd({
   *       options: {
   *         adUnitId: AD_UNIT_ID,
   *       },
   *       onEvent: (event) => {
   *         switch (event.type) {
   *           case 'loaded':
   *             console.log('광고 로드 성공', event.data);
   *             break;
   *
   *           case 'clicked':
   *             console.log('광고 클릭');
   *             break;
   *
   *           case 'dismissed':
   *             console.log('광고 닫힘');
   *             break;
   *
   *           case 'failedToShow':
   *             console.log('광고 보여주기 실패');
   *             break;
   *
   *           case 'impression':
   *             console.log('광고 노출');
   *             break;
   *
   *           case 'show':
   *             console.log('광고 컨텐츠 보여졌음');
   *             break;
   *         }
   *       },
   *       onError: (error) => {
   *         console.error('광고 불러오기 실패', error);
   *       },
   *     });
   *
   *     return cleanup;
   *   }, []);
   *
   *   return (
   *     <View>
   *       <Text>Page</Text>
   *     </View>
   *   );
   * }
   * ```
   */
  loadAdMobInterstitialAd: Object.assign(
    createEventBridge("loadAdMobInterstitialAd"),
    {
      isSupported: createConstantBridge2("loadAdMobInterstitialAd_isSupported")
    }
  ),
  /**
   * @public
   * @category 광고
   * @name showAdMobInterstitialAd
   * @description 앱 화면 전체를 덮는 전면 광고를 사용자에게 노출해요. 이 함수는 `loadAdMobInterstitialAd` 로 미리 불러온 광고를 실제로 사용자에게 노출해요.
   * @param {ShowAdMobInterstitialAdParams} params 광고를 보여주기 위해 사용할 설정 값이에요. 광고 ID와과 광고의 동작에 대한 콜백을 설정할 수 있어요.
   * @param {ShowAdMobInterstitialAdOptions} params.options 광고를 보여줄 때 전달할 옵션 객체예요.
   * @param {string} params.options.adUnitId 광고 단위 ID예요. `loadAdMobInterstitialAd` 로 불러온 전면 광고용 ID를 입력해요.
   * @param {(event: ShowAdMobInterstitialAdEvent) => void} [params.onEvent] 광고 관련 이벤트가 발생했을 때 호출돼요. (예시: 광고 노출을 요청했을 때). 자세한 이벤트 타입은 [ShowAdMobInterstitialAdEvent](/react-native/reference/framework/광고/ShowAdMobInterstitialAdEvent.html)를 참고하세요.
   * @param {(reason: unknown) => void} [params.onError] 광고를 노출하지 못했을 때 호출돼요. (예시: 네트워크 오류나 지원하지 않는 환경일 때)
   * @property {() => boolean} [isSupported] 현재 실행 중인 앱(예: 토스 앱, 개발용 샌드박스 앱 등)에서 Google AdMob 광고 기능을 지원하는지 확인하는 함수예요. 기능을 사용하기 전에 지원 여부를 확인해야 해요.
   *
   * @example
   * ### 버튼 눌러 불러온 전면 광고 보여주기
   * ```tsx
   * import { GoogleAdMob } from '@apps-in-toss/framework';
   * import { View, Text, Button } from 'react-native';
   *
   * const AD_UNIT_ID = '<YOUR_AD_UNIT_ID>';
   *
   * function Page() {
   *   const handlePress = () => {
   *     if (GoogleAdMob.showAdMobInterstitialAd.isSupported() !== true) {
   *       return;
   *     }
   *
   *     GoogleAdMob.showAdMobInterstitialAd({
   *       options: {
   *         adUnitId: AD_UNIT_ID,
   *       },
   *       onEvent: (event) => {
   *         switch (event.type) {
   *           case 'requested':
   *             console.log('광고 보여주기 요청 완료');
   *             break;
   *         }
   *       },
   *       onError: (error) => {
   *         console.error('광고 보여주기 실패', error);
   *       },
   *     });
   *   }
   *
   *   return (
   *     <Button onPress={handlePress} title="광고 보기" />
   *   );
   * }
   * ```
   */
  showAdMobInterstitialAd: Object.assign(
    createEventBridge("showAdMobInterstitialAd"),
    {
      isSupported: createConstantBridge2("showAdMobInterstitialAd_isSupported")
    }
  ),
  /**
   * @public
   * @category 광고
   * @name loadAdMobRewardedAd
   * @description 사용자가 광고를 끝까지 시청하면 리워드를 제공할 수 있는 보상형 광고를 미리 불러와서, 광고가 필요한 시점에 바로 보여줄 수 있도록 준비하는 함수예요.
   * @param {LoadAdMobRewardedAdParams} params 광고를 불러올 때 사용할 설정 값이에요. 광고 ID와 광고의 동작에 대한 콜백을 설정할 수 있어요.
   * @param {LoadAdMobRewardedAdOptions} params.options 광고를 불러올 때 전달할 옵션 객체예요.
   * @param {string} params.options.adUnitId 광고 단위 ID예요. 발급받은 보상형 광고용 ID를 입력해요.
   * @param {(event: LoadAdMobRewardedAdEvent) => void} [params.onEvent] 광고 관련 이벤트가 발생했을 때 호출돼요. (예시: 광고가 닫히거나 클릭됐을 때). 자세한 이벤트 타입은 [LoadAdMobRewardedAdEvent](/react-native/reference/framework/광고/LoadAdMobRewardedAdEvent.html)를 참고하세요.
   * @param {(reason: unknown) => void} [params.onError] 광고를 불러오지 못했을 때 호출돼요. (예시: 네트워크 오류나 지원하지 않는 환경일 때)
   * @property {() => boolean} [isSupported] 현재 실행 중인 앱(예: 토스 앱, 개발용 샌드박스 앱 등)에서 Google AdMob 광고 기능을 지원하는지 확인하는 함수예요. 기능을 사용하기 전에 지원 여부를 확인해야 해요.
   *
   * @example
   * ### 뷰 진입 시 보상형 광고 불러오기
   * ```tsx
   * import { GoogleAdMob } from '@apps-in-toss/framework';
   * import { useEffect } from 'react';
   * import { View, Text } from 'react-native';
   *
   * const AD_UNIT_ID = '<YOUR_AD_UNIT_ID>';
   *
   * function Page() {
   *   useEffect(() => {
   *     if (GoogleAdMob.loadAdMobRewardedAd.isSupported() !== true) {
   *       return;
   *     }
   *
   *     const cleanup = GoogleAdMob.loadAdMobRewardedAd({
   *       options: {
   *         adUnitId: AD_UNIT_ID,
   *       },
   *       onEvent: (event) => {
   *         switch (event.type) {
   *           case 'loaded':
   *             console.log('광고 로드 성공', event.data);
   *             break;
   *
   *           case 'clicked':
   *             console.log('광고 클릭');
   *             break;
   *
   *           case 'dismissed':
   *             console.log('광고 닫힘');
   *             break;
   *
   *           case 'failedToShow':
   *             console.log('광고 보여주기 실패');
   *             break;
   *
   *           case 'impression':
   *             console.log('광고 노출');
   *             break;
   *
   *           case 'show':
   *             console.log('광고 컨텐츠 보여졌음');
   *             break;
   *
   *           case 'userEarnedReward':
   *             console.log('사용자가 광고 시청을 완료했음');
   *             break;
   *         }
   *       },
   *       onError: (error) => {
   *         console.error('광고 불러오기 실패', error);
   *       },
   *     });
   *
   *     return cleanup;
   *   }, []);
   *
   *   return (
   *     <View>
   *       <Text>Page</Text>
   *     </View>
   *   );
   * }
   * ```
   */
  loadAdMobRewardedAd: Object.assign(
    createEventBridge("loadAdMobRewardedAd"),
    {
      isSupported: createConstantBridge2("loadAdMobRewardedAd_isSupported")
    }
  ),
  /**
   * @public
   * @category 광고
   * @name showAdMobRewardedAd
   * @description 사용자가 광고를 끝까지 보면 리워드를 받을 수 있도록, 보상형 광고를 화면에 보여줘요. 이 함수는 `loadAdMobRewardedAd` 로 미리 불러온 광고를 실제로 사용자에게 노출해요.
   * @param {ShowAdMobRewardedAdParams} params 광고를 보여주기 위해 사용할 설정 값이에요. 광고 ID와 광고의 동작에 대한 콜백을 설정할 수 있어요.
   * @param {ShowAdMobRewardedAdOptions} params.options 광고를 보여줄 때 전달할 옵션 객체예요.
   * @param {string} params.options.adUnitId 광고 단위 ID예요. `loadAdMobRewardedAd` 로 불러온 보상형 광고용 ID를 입력해요.
   * @param {(event: ShowAdMobRewardedAdEvent) => void} [params.onEvent] 광고 관련 이벤트가 발생했을 때 호출돼요. (예시: 광고 노출을 요청했을 때). 자세한 이벤트 타입은 [ShowAdMobRewardedAdEvent](/react-native/reference/framework/광고/ShowAdMobRewardedAdEvent.html)를 참고하세요.
   * @param {(reason: unknown) => void} [params.onError] 광고를 불러오지 못했을 때 호출돼요. (예시: 네트워크 오류나 지원하지 않는 환경일 때)
   * @property {() => boolean} [isSupported] 현재 실행 중인 앱(예: 토스 앱, 개발용 샌드박스 앱 등)에서 Google AdMob 광고 기능을 지원하는지 확인하는 함수예요. 기능을 사용하기 전에 지원 여부를 확인해야 해요.
   *
   * @example
   * ### 버튼 눌러 불러온 보상형 광고 보여주기
   * ```tsx
   * import { GoogleAdMob } from '@apps-in-toss/framework';
   * import { View, Text, Button } from 'react-native';
   *
   * const AD_UNIT_ID = '<YOUR_AD_UNIT_ID>';
   *
   * function Page() {
   *   const handlePress = () => {
   *     if (GoogleAdMob.showAdMobRewardedAd.isSupported() !== true) {
   *       return;
   *     }
   *
   *     GoogleAdMob.showAdMobRewardedAd({
   *       options: {
   *         adUnitId: AD_UNIT_ID,
   *       },
   *       onEvent: (event) => {
   *         switch (event.type) {
   *           case 'requested':
   *             console.log('광고 보여주기 요청 완료');
   *             break;
   *         }
   *       },
   *       onError: (error) => {
   *         console.error('광고 보여주기 실패', error);
   *       },
   *     });
   *   }
   *
   *   return (
   *     <Button onPress={handlePress} title="광고 보기" />
   *   );
   * }
   * ```
   */
  showAdMobRewardedAd: Object.assign(
    createEventBridge("showAdMobRewardedAd"),
    {
      isSupported: createConstantBridge2("showAdMobRewardedAd_isSupported")
    }
  )
};

// src/bedrockEvent.ts
import { createEventBridge as createEventBridge2 } from "@apps-in-toss/bridge-core";
var bedrockEvent = {
  addEventListener: (event, {
    onEvent,
    onError,
    options
  }) => createEventBridge2(event)({
    onEvent,
    onError: onError ?? (() => {
    }),
    options
  })
};

// src/isMinVersionSupported.ts
import { createConstantBridge as createConstantBridge3 } from "@apps-in-toss/bridge-core";

// src/utils/compareVersion.ts
var SEMVER_REGEX = /^[v^~<>=]*?(\d+)(?:\.([x*]|\d+)(?:\.([x*]|\d+)(?:\.([x*]|\d+))?(?:-([\da-z\\-]+(?:\.[\da-z\\-]+)*))?(?:\+[\da-z\\-]+(?:\.[\da-z\\-]+)*)?)?)?$/i;
var isWildcard = (val) => ["*", "x", "X"].includes(val);
var tryParse = (val) => {
  const num = parseInt(val, 10);
  return isNaN(num) ? val : num;
};
var coerceTypes = (a, b) => {
  return typeof a === typeof b ? [a, b] : [String(a), String(b)];
};
var compareValues = (a, b) => {
  if (isWildcard(a) || isWildcard(b)) {
    return 0;
  }
  const [aVal, bVal] = coerceTypes(tryParse(a), tryParse(b));
  if (aVal > bVal) {
    return 1;
  }
  if (aVal < bVal) {
    return -1;
  }
  return 0;
};
var parseVersion = (version) => {
  if (typeof version !== "string") {
    throw new TypeError("Invalid argument: expected a string");
  }
  const match = version.match(SEMVER_REGEX);
  if (!match) {
    throw new Error(`Invalid semver: '${version}'`);
  }
  const [, major, minor, patch, build, preRelease] = match;
  return [major, minor, patch, build, preRelease];
};
var compareSegments = (a, b) => {
  const maxLength = Math.max(a.length, b.length);
  for (let i = 0; i < maxLength; i++) {
    const segA = a[i] ?? "0";
    const segB = b[i] ?? "0";
    const result = compareValues(segA, segB);
    if (result !== 0) {
      return result;
    }
  }
  return 0;
};
var compareVersions = (v1, v2) => {
  const seg1 = parseVersion(v1);
  const seg2 = parseVersion(v2);
  const preRelease1 = seg1.pop();
  const preRelease2 = seg2.pop();
  const mainCompare = compareSegments(seg1, seg2);
  if (mainCompare !== 0) {
    return mainCompare;
  }
  if (preRelease1 && preRelease2) {
    return compareSegments(preRelease1.split("."), preRelease2.split("."));
  }
  if (preRelease1) {
    return -1;
  }
  if (preRelease2) {
    return 1;
  }
  return 0;
};

// src/isMinVersionSupported.ts
function isMinVersionSupported(minVersions) {
  const operationalEnvironment = createConstantBridge3("getOperationalEnvironment")();
  if (operationalEnvironment === "sandbox") {
    return true;
  }
  const currentVersion = createConstantBridge3("getTossAppVersion")();
  const isIOS = createConstantBridge3("getPlatformOS")() === "ios";
  const minVersion = isIOS ? minVersions.ios : minVersions.android;
  if (minVersion === void 0) {
    return false;
  }
  if (minVersion === "always") {
    return true;
  }
  if (minVersion === "never") {
    return false;
  }
  return compareVersions(currentVersion, minVersion) >= 0;
}

// src/appsInTossEvent.ts
import { createEventBridge as createEventBridge3 } from "@apps-in-toss/bridge-core";
var appsInTossEvent = {
  addEventListener: (event, {
    onEvent,
    onError,
    options
  }) => createEventBridge3(event)({
    onEvent,
    onError: onError ?? (() => {
    }),
    options
  })
};

// src/env.ts
import { createConstantBridge as createConstantBridge4 } from "@apps-in-toss/bridge-core";
var env = {
  getDeploymentId: createConstantBridge4("getDeploymentId")
};

// src/global.ts
import { createConstantBridge as createConstantBridge5 } from "@apps-in-toss/bridge-core";
var deploymentId = createConstantBridge5("deploymentId");
var brandDisplayName = createConstantBridge5("brandDisplayName");
var brandIcon = createConstantBridge5("brandIcon");
var brandPrimaryColor = createConstantBridge5("brandPrimaryColor");
var brandBridgeColorMode = createConstantBridge5("brandBridgeColorMode");
var getAppsInTossGlobals = () => {
  return {
    deploymentId: deploymentId(),
    brandDisplayName: brandDisplayName(),
    brandIcon: brandIcon(),
    brandPrimaryColor: brandPrimaryColor(),
    brandBridgeColorMode: brandBridgeColorMode()
  };
};

// src/tdsEvent.ts
import { createEventBridge as createEventBridge4 } from "@apps-in-toss/bridge-core";
var tdsEvent = {
  /**
   * @public
   * @category TDS
   * @name addEventListener
   * @description 이벤트를 추가해요.
   * @param {string} event 이벤트 이름예요.
   * @param {TdsEvent[K]['onEvent']} onEvent 이벤트 핸들러예요.
   * @param {TdsEvent[K]['onError']} onError 이벤트 에러 핸들러예요.
   * @param {TdsEvent[K]['options']} options 이벤트 옵션예요.
   * @returns {void} 이벤트를 추가했을 때 아무 값도 반환하지 않아요.
   * @example
   * ### 네비게이션 악세서리 이벤트 추가하기
   * ```tsx
   * import { tdsEvent } from '@apps-in-toss/framework';
   *
   * tdsEvent.addEventListener('navigationAccessoryEvent', {
   *   onEvent: ({ id }) => {
   *     if(id === 'heart') {
   *       router.push('/heart');
   *     }
   *   },
   * });
   * ```
   */
  addEventListener: (event, {
    onEvent,
    onError,
    options
  }) => createEventBridge4(event)({
    onEvent,
    onError: onError ?? (() => {
    }),
    options
  })
};

// src/partner.ts
import { createAsyncBridge as createAsyncBridge3 } from "@apps-in-toss/bridge-core";
var partner = {
  /**
   * @public
   * @category 파트너
   * @name addAccessoryButton
   * @description 상단 네비게이션의 악세서리 버튼을 추가해요. callback에 대한 정의는 `tdsEvent.addEventListener("navigationAccessoryEvent", callback)`를 참고해주세요.
   * @param {AddAccessoryButtonOptions} options - 악세서리 버튼의 고유 ID예요.
   * @returns {void} 악세서리 버튼을 추가했을 때 아무 값도 반환하지 않아요.
   * @example
   * ```tsx
   * import { partner } from '@apps-in-toss/framework';
   *
   * partner.addAccessoryButton({
   *   id: 'heart',
   *   type: 'icon',
   *   name: 'icon-heart-mono',
   *   alt: '하트',
   * });
   *
   * partner.addAccessoryButton({
   *   id: ['heart', 'heart2'],
   *   type: 'icon-double',
   *   leftIconName: 'icon-heart-mono',
   *   leftIconAlt: '하트',
   *   rightIconName: 'icon-heart-mono',
   *   rightIconAlt: '하트',
   * });
   * ```
   */
  addAccessoryButton: createAsyncBridge3("addAccessoryButton"),
  /**
   * @public
   * @category 파트너
   * @name removeAccessoryButton
   * @description 상단 네비게이션의 악세서리 버튼을 제거해요.
   * @returns {void} 악세서리 버튼을 제거했을 때 아무 값도 반환하지 않아요.
   * @example
   * ```tsx
   * import { partner } from '@apps-in-toss/framework';
   *
   * partner.removeAccessoryButton();
   * ```
   */
  removeAccessoryButton: createAsyncBridge3("removeAccessoryButton")
};
export {
  GoogleAdMob,
  IAP,
  Storage,
  appsInTossEvent,
  bedrockEvent,
  env,
  getAppsInTossGlobals,
  getSafeAreaInsets,
  isMinVersionSupported,
  partner,
  tdsEvent
};
